website_oca_integrator/models/res_partner.py

```diff
@@ -0,0 +1,281 @@
+                    commit_url = pull_request['head']['repo']['commits_url']\
+                        .replace('{/sha}', '/' + commit_sha)
+                    try:
+                        commit_data = github_connector.get_by_url(
+                            commit_url, 'get')
+                    except Exception:
+                        _logger.warning(
+                            "Error while calling url '%s' while "
+                            "fetching module for '%s'." %
+                            (pull_request['url'], x['actor']['login']))
+                        continue
+
+                    # find module name(s) from all committed files
+                    for commit_file in commit_data['files']:
+                        if len(current_page_modules) + all_modules_len >= 5:
+                            return current_page_modules
+                        file_name = commit_file.get(
+                            'filename').split('/')[0].split('.')
+                        # if length of filename is 1 then filename
+                        # can be technical name of module
+                        if len(file_name) == 1 and not \
+                                file_name[0] in current_page_modules:
+                            module_name = file_name
+                            odoo_module = self.env['odoo.module'].search(
+                                [('technical_name', 'in', module_name)])
+                            if odoo_module:
+                                date = datetime.datetime.strptime(
+                                    pr_data['merged_at'],
+                                    "%Y-%m-%dT%H:%M:%SZ").strftime(
+                                    '%Y-%m-%d %H:%M:%S')
+                                current_page_modules[module_name[0]] = date
+
+        return current_page_modules
+
+    def contributors_fetch_modules(self):
+        all_modules = {}
+
+        contributors = self.env['res.partner'].search(
+            ['&', ('github_login', '!=', False),
+             ('website_published', '=', True)])
+        github_connector = self.env[
+            'abstract.github.model'].get_github_connector('user')
+
+        for record in contributors:
+            contributor_login = record.github_login + '/events'
+            # github API is limited to only 10 pages.
+            for page in range(1, 11):
+                try:
+                    user_data = github_connector.get([contributor_login],
+                                                     page=page)
+                except Exception:
+                    _logger.warning("Github login for partner '%s' is not"
+                                    "correctly set." % (record.name))
+                    break
+                res = self.get_github_user_modules(
+                    user_data, github_connector, len(all_modules))
+                all_modules.update(res)
+                if len(all_modules) == 5:
```

Comment:
Maybe you are saying because of this line.
https://github.com/OCA/oca-custom/blob/3ba74cde2048fb7e26a608e10ec77444091e8183/website_oca_integrator/models/res_partner.py#L233

If i change above line to ‘=’, then I guess we don’t need to change as we have to fetch only 5 modules.